catalog/imaging/imaging-transformer-nitf/src/test/groovy/NitfPreStoragePluginTest.groovy

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+import spock.lang.*
+import org.codice.alliance.transformer.nitf.image.NitfPreStoragePlugin
+
+@Unroll
+class NitfPreStoragePluginTest extends Specification {
+
+    def "Building derived image filename from \"#ftitle\"" (String ftitle, String expectedFname) {
+        setup:
+            NitfPreStoragePlugin plugin = new NitfPreStoragePlugin();
+            def qualifier = "original";
+
+        when: "building a derived image filename"
+            def derivedFname = plugin.buildDerivedImageTitle(ftitle, qualifier);
+
+        then: "the derived filenames should not include invalid characters"
+            derivedFname.equals(expectedFname)
```

Comment:
Looks so much better now!

One Groovy tip: you can use `==` for equality tests.